code/missionui/chatbox.cpp

```diff
@@ -433,22 +454,16 @@ void chatbox_autosplit_line()
 		} else {
 			remainder = "";
 		}	
-		// if I'm the server, then broadcast the packet		
-		chatbox_recall_add(msg);
+		
+		chatbox_send_msg(msg);
 
-		if (target != MULTI_MSG_EXPR) {
-  			send_game_chat_packet(Net_player, msg, target);
-		}
-		else {
-			// copy the name of the player the message is being sent to
-			strncpy(temp, msg+1, target_length-2);
-			temp[target_length-2] = '\0';
-			send_game_chat_packet(Net_player, msg, target, NULL, temp);
-		}
-		chatbox_add_line(msg, MY_NET_PLAYER_NUM);
+		// check if this message is supposed to have a recipient
+		int target = chatbox_get_msg_target_type(msg);
+		size_t target_length = chatbox_get_msg_target_length(msg);
 
 		if (target != MULTI_MSG_ALL) {
 			// we need to add the target the message is going to before we add the rest of the string
+			char temp[150];
 			strncpy(temp, msg, target_length);
  			temp[target_length] = ' '; 
  			temp[target_length+1] = '\0'; 
```

Comment:
As mentioned in the earlier change request, the size of `temp` should be de-magic-numbered here, as well as moving `target_length`.